analyzers/src/SonarAnalyzer.Common/Rules/ClassNotInstantiatableBase.cs

```diff
@@ -31,20 +31,18 @@ public abstract class ClassNotInstantiatableBase : SonarDiagnosticAnalyzer
         protected const string MessageFormat = "This {0} can't be instantiated; make {1} 'public'.";
 
         protected abstract bool IsTypeDeclaration(SyntaxNode node);
+        protected abstract bool IsObjectCreation(SyntaxNode node);
```

Comment:
Let's say you provide `ILanguageFacade` instead to this class. Like that, you can use `IsAnyKind` (from the facade) with the `ObjectCreationSyntaxKinds`.